packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2271,6 +2274,16 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     // sending multiple `TextInput.updateEditingValue` messages.
     beginBatchEdit();
     _value = value;
+    // Changes made by the keyboard can sometimes be "out of band" for listening
+    // components, so always send those events, even if we didn't think it
+    // changed. Also, focusing an empty field is sent as a selection change even
+    // if the selection offset didn't change.
+    if (selectionChanged ||
+        (userInteraction &&
+        ((value.selection.baseOffset == 0 && value.selection.extentOffset == 0 && !_hasFocus) ||
```

Comment:
this is for corner case where the textfield is active and user single tap the textfield again to show the selection overlay.
Therefore, we only want to build overlay when it is trigger by the user if the selection is still at 0

Reply:
ah yea i should check for selection change before calling onselectionchange

Reply:
why the `!_hasFocus` if the context menu should only be updated on a focused field?

Reply:
ok this is a series of legacy code and bug make me think this is correct. We want to build the selection overlay when user type keyboard or long press. and we need to want to call the onselectionchange because textfield relies on it to decide whether it really want to show selection handles. (I know it is weird, but again, this why i think we should move selectionoverlay outside..). It has already like this prior to the change, that longpress on a textfield will fire selection change even if selection does not change.

Reply:
I update it to check for long press instead